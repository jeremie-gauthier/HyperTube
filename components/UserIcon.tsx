import { TUser } from "@/data/models/User";
import { FlexCol } from "@/components/Flex";
import { getInitials } from "@/lib/helpers";
import styles from "./UserIcon.module.scss";

type UserIconProps = {
  user: TUser;
};

export default function UserIcon({
  user,
  ...rest
}: UserIconProps): JSX.Element {
  return (
    <FlexCol className={styles.frame} {...rest}>
      <span>{getInitials(user)}</span>
    </FlexCol>
  );
}
