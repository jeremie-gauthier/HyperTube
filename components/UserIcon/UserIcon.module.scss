.frame {
  @apply items-center justify-center rounded h-8 w-8 cursor-pointer bg-red;

  span {
    @apply font-bold text-sm;
  }
}
