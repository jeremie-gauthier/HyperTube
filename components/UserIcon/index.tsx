import { User } from "@/types/user";
import { FlexCol } from "@/components/Flex";
import Image from "next/image";
import styles from "./UserIcon.module.scss";

type UserIconProps = React.HTMLAttributes<HTMLDivElement> & {
  user: User;
};

export default function UserIcon({ user, ...rest }: UserIconProps) {
  const { picture } = user;

  return (
    <FlexCol {...rest} className={styles.frame}>
      {picture && (
        <Image
          src={`/img/avatar/avatar${user.picture}.png`}
          alt="Current profile picture"
          width={32}
          height={32}
          quality={100}
          key={user.picture}
        />
      )}
    </FlexCol>
  );
}
