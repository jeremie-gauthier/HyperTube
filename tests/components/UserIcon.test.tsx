import React from "react";
import { render, screen } from "@testing-library/react";
import UserIcon from "@/components/UserIcon";
import userMock from "@/tests/__mocks__/user";

test("renders correctly", () => {
  const { container } = render(<UserIcon user={userMock} />);
  expect(container).toMatchSnapshot();
});

test("expect the initials to be visible", () => {
  render(<UserIcon user={userMock} />);

  expect(screen.getByText(/AD/i)).toBeInTheDocument();
});
