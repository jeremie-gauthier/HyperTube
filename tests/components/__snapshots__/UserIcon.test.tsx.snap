// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    class="flexCol frame"
  >
    <span>
      AD
    </span>
  </div>
</div>
`;
